app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -60,7 +60,7 @@ def plan(job_invocation, host, template_invocation, proxy_selector = ::RemoteExe
                                :use_concurrency_control => options[:use_concurrency_control],
                                :first_execution => first_execution,
                                :alternative_names => provider.alternative_names(host) }
-        action_options = provider.proxy_command_options(template_invocation, host)
+        action_options = provider.proxy_command_options(template_invocation, host, job_invocation)
```

Comment:
This is slightly problematic. Other plugins implement their own providers (such as [ansible](https://github.com/theforeman/foreman_ansible/blob/3660359a5afb3c9d71b1e52b4c05366881835427/app/models/foreman_ansible/ansible_provider.rb#L25)), but this change changes the interface we expect the providers to implement. While we can technically do that, I'd like to avoid that if possible.

Here we don't really need to pass the job invocation into the function, we can just pull the job invocation out of the template invocation in that function.

Reply:
Yes, that's true. Thanks for suggestion!